### PR TITLE
changed plot.py to work with notebooks.

### PR DESCRIPTION
**Pull request by JoyMonteiro**
As discussed in #3, this PR enables PlotFunctionMonitor to work with notebooks and the command line.

It also adds a `plt.tight_layout()` command to make figures look neater.

### CODE REVIEW

File: sympl/_components/plot.py [hunk 1]
```diff
@@ -64,13 +63,13 @@ def store(self, state):
...
             self._fig.clear()
-            time.sleep(1e-5)
             fig = self._fig
         else:
             fig = plt.figure()
         self._plot_function(fig, copy_state(state))
-        plt.draw_all()
+        plt.tight_layout()
+        fig.canvas.draw()
         if self.interactive:
```
Comment (mcgibbon) [inline sympl/_components/plot.py]: We should refactor this to use `if not self.interactive`, to avoid the `pass` block.

**Comment by mcgibbon** [comment]
I suggested one small change, after that this can be merged.

**Comment by JoyMonteiro** [comment]
done!

File: sympl/_components/plot.py [hunk 2]
```diff
@@ -64,13 +63,11 @@ def store(self, state):
...
-            time.sleep(1e-5)
             fig = self._fig
         else:
             fig = plt.figure()
         self._plot_function(fig, copy_state(state))
-        plt.draw_all()
-        if self.interactive:
-            time.sleep(1e-5)  # necessary to draw, pause can be arbitrarily small
-        else:
+        plt.tight_layout()
```
Comment (mcgibbon) [inline sympl/_components/plot.py]: One more change, didn't notice this before. plt.tight_layout() should be given by the user in the plot function if they want it, so it shouldn't be called here. Also plt.tight_layout isn't guaranteed to apply tight layout to the figure used in this object, if other figures have been initialized after it. So I think we should remove this line.

**Comment by JoyMonteiro** [comment]
fair enough, deleted...

**Pull request merged**